### ISSUE TITLE
[Problem] Materials: Please support import/export MTL file (Material Template Library format) and OBJ alone or together

### ISSUE THREAD

**Issue by supermangithu** [FreeCAD/FreeCAD#11887]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

 Please support import/export mtl file(Material Template Library format ) and obj alone or together!
It's been a long time since this problem did not solve.
since https://forum.freecad.org/viewtopic.php?style=1&t=44803

### Full version info

```shell
OS: Windows 10 build 14393
Word size of FreeCAD: 64-bit
Version: 0.21.1.33668 +26 (Git)
Build type: Release
Branch: (HEAD detached at 0.21.1)
Hash: f6708547a9bb3f71a4aaade12109f511a72c207c
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.3
Locale: Chinese/China (zh_CN)
```

### Subproject(s) affected?

None

### Anything else?

 https://forum.freecad.org/viewtopic.php?style=1&t=44803

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
@davesrocketshop is this possible to consider with the material overhaul? I guess this file only has appearence parameters.

**Comment by davesrocketshop** [comment]
> @davesrocketshop is this possible to consider with the material overhaul? I guess this file only has appearence parameters.

Yes, but it would be a low priority. Most likely for the next release.

**Comment by davesrocketshop** [comment]
> > > @davesrocketshop is this possible to consider with the material overhaul? I guess this file only has appearence parameters.
> > 
> > 
> > Yes, but it would be a low priority. Most likely for the next release.
> 
> I hope it can be really solved in next version release.Probaly 0.22?Thanks in advance! Also,I hope your file .FCStd can be a open standard that let those CATIA ,UG ...,etc support to export Graphics with full parameters (like mesured dimensions,Spline with equation or NURBS with parameters).I personally think Freecad developers ought to interactive with developers ,such as CATIA's in acedemic / industrial conferences.Maybe come true?

I guess my answer wasn't clear. It definitely will not be in 0.22. There are too many higher priority items.

**Comment by raw915** [comment]
It appears that FreeCAD names each object's material according to the single color assigned to it. Thus, if one body has color #ff0000 and a second body in the same exported .obj file also has color #ff0000, nothing will be written into the .mtl file for the second body. Change one of the colors to #ff0001 or something similar, and c_ff0000 and c_ff0001 will be written into the .mtl file and both bodies will have proper color display.

**Comment by davesrocketshop** [comment]
Materials have UUIDs associated with them.

**Comment by raw915** [comment]
> Materials have UUIDs associated with them.

What does that mean? Here are the contents of my .mtl file:
# FreeCAD v0.21 build2 Arch module
# https://www.freecad.org
newmtl color_ff0000
Kd 1.0 0.0 0.0
Tr 0.0
d 1.0
newmtl color_00ff00
Kd 0.0 1.0 0.0
Tr 0.0
d 1.0
newmtl color_0000ff
Kd 0.0 0.0 1.0
Tr 0.0
d 1.0
newmtl color_ff0001
Kd 1.0 0.0 0.003921568859368563
Tr 0.9
d 0.09999999999999998
# Material Count: 4

A red body, a green body, a blue body, and another almost red body. This works as expected. But if I make the last body exactly red, its name color_ff0000 will conflict with the first color_ff0000 name and no color information will be written into the .mtl file for the last body. That body will be given a color_cccccc value and come out gray, clearly not what I want. Is that a bug or me not knowing how to properly export colored bodies to .obj format?

**Comment by luzpaz** [comment]
@supermangithu Materials have been rewritten in v0.22 so if you want to troubleshoot then use the development version.

**Comment by m1cha1** [comment]
MTL file is supported by Freecad but with bugs (For example here: https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Mesh/App/Core/IO/ReaderOBJ.cpp). @maxwxyz Could you please change the Label.
